src/protosext/mod.rs

```diff
@@ -302,13 +302,25 @@ pub struct ValidScheduleLA {
     pub cancellation_type: ActivityCancellationType,
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Copy)]
 pub enum LACloseTimeouts {
     ScheduleOnly(Duration),
     StartOnly(Duration),
     Both { sched: Duration, start: Duration },
 }
 
+impl LACloseTimeouts {
+    /// Splits into (schedule_to_close, start_to_close) options, one or both of which is guaranteed
+    /// to be populated
```

Comment:
bruh that is this type :laughing:

Reply:
Nah, it's not... you **obviously** need a union return type:
`(Some<Duration>, Some<Duration>) | (Some<Duration>, None) | (None, Some<Duration>)`

Reply:
Call me old fashioned, but `pub struct LACloseTimeouts { sched: Option<Duration>, start: Option<Duration> }` with a runtime check to fail both as not present is my preference for simplicity.

Reply:
@cretz That's exactly what this type represents. There just happens to be a need to split the type into the two options for convenience reasons, but it can't be *constructed* without at least one being present.

Reply:
Right, I was just saying I like the single struct over the enum personally (I like not having to match the three enum options and I accept the cost of runtime check and no compile-time check for both being absent).